File: apple/Sources/ReactNativeWatchOS/Modifiers/PresentationCompactAdaptation.swift
import SwiftUI

/// SwiftUI `.presentationCompactAdaptation(_:)`.
///
/// LIMITATION: `presentationCompactAdaptation` and the
/// `PresentationAdaptation` type are **iOS 16.4+ / macOS 13.3+ / tvOS
/// only — not available on watchOS** (watchOS has a single compact size
/// class with no adaptive presentation styles, and the symbol cannot be
/// referenced here). This modifier is registered as a no-op: it accepts
/// the `adaptation` param for cross-platform JS parity and returns the
/// view unchanged.
enum RNWPresentationCompactAdaptationModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("presentationCompactAdaptation") { view, _, _ in
            // No-op on watchOS — `presentationCompactAdaptation` is unavailable.
            view
        }
    }
}

File: src/modifiers/presentationCompactAdaptation.ts
import { createModifier } from './createModifier';

/// How a presentation adapts to a compact size class. Mirrors SwiftUI's
/// `PresentationAdaptation`.
export type PresentationAdaptation =
  | 'automatic'
  | 'none'
  | 'popover'
  | 'sheet'
  | 'fullScreenCover';

/// SwiftUI `.presentationCompactAdaptation(_:)`. Specifies how a
/// presentation should adapt when running in a horizontally and vertically
/// compact size class. Defaults to `'automatic'`.
///
/// NOTE: `presentationCompactAdaptation` is **not available on watchOS**
/// (iOS 16.4+ / macOS 13.3+ only). On watchOS this modifier is a
/// documented no-op — accepted for cross-platform JS parity, applies
/// nothing.
export function presentationCompactAdaptation(
  adaptation: PresentationAdaptation = 'automatic'
) {
  return createModifier('presentationCompactAdaptation', { adaptation });
}
